File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day13Test.java
package com.capital7software.aoc2015.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.aoc2015.days.Day13;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day13Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day13Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day13();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void greatestChangeInHappiness() {
    var instance = new Day13();
    var expected = 330;
    var actual = instance.greatestChangeInHappiness(lines);
    assertEquals(
        expected,
        actual,
        "Greatest change in happiness is not what was expected: " + expected
    );
  }

  @Test
  void greatestChangeInHappinessWithMeAtTheParty() {
    var instance = new Day13();
    var expected = 286;
    setupFromFile("inputs/input_day_13-02.txt");
    var actual = instance.greatestChangeInHappiness(lines);
    assertEquals(
        expected,
        actual,
        "Greatest change in happiness is not what was expected: " + expected
    );
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day13Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day13Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day13Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day13()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testTakeTrip() {
    val instance = Day13()
    val expected = Pair(0, 24)

    val actual = instance.takeTrip(lines)

    Assertions.assertEquals(
        expected, actual, "The severity " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testTakeTripNoCatch() {
    val instance = Day13()
    val expected = Pair(10, 0)

    val actual = instance.takeTrip(lines, true)

    Assertions.assertEquals(
        expected, actual, "The severity " +
        "is not what was expected: $expected"
    )
  }
}
